### PR TITLE
[CU-2vzagvv] Split linter checks to separate CI jobs

### PR DESCRIPTION
**Pull request by andor0**
## Issue
https://app.clickup.com/t/2vzagvv

## Description
Also, I added `nixfmt` check.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2vzagvv Split linter checks to separate CI jobs](https://app.clickup.com/t/2vzagvv)

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -114,31 +114,195 @@ jobs:
...
-      - name: Cargo fmt
-        run: |
+      - run: |
           cachix watch-exec -j 16 -c 0 composable-community nix -- build .#cargo-fmt-check --no-update-lock-file --show-trace -L
 
-      - name: Taplo CLI
-        run: |
+  taplo-cli-check:
+    name: "Taplo cli check"
+    needs: common-deps
```
Comment (KaiserKarel) [inline .github/workflows/build-test-deploy.yml:122]: I think taplo does not need common deps?

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -114,31 +114,195 @@ jobs:
...
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          name: composable-community
+      - run: |
           cachix watch-exec -j 16 -c 0 composable-community nix -- build .#prettier-check --no-update-lock-file --show-trace -L
 
-      - name: Cargo clippy
-        run: |
+  nixfmt-check:
+    name: "Nixfmt check"
+    needs: common-deps
```
Comment (KaiserKarel) [inline .github/workflows/build-test-deploy.yml:184]: Probably doesn't need common checks either?

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml:122]: yes, you are right.

File: .github/workflows/build-test-deploy.yml (hunk 2 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml:184]: yes, you are right.
and `prettier` also doesn't require `common-deps`.

**Review by dzmitry-lahoda — APPROVED** [review]

File: .github/workflows/build-test-deploy.yml (hunk 1 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml]: fixed.

File: .github/workflows/build-test-deploy.yml (hunk 2 above)
Comment (andor0) [reply .github/workflows/build-test-deploy.yml]: fixed.

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Review by Dom-Roth — APPROVED** [review]

**Comment by andor0** [comment]
@Mergifyio refresh

**Pull request merged**